src/cplscheme/CouplingData.hpp

```diff
@@ -45,14 +47,24 @@ class CouplingData {
   ///  True, if the data values of this CouplingData require to be initialized by this participant.
   const bool requiresInitialization;
 
+  /// initialize _extrapolation
+  void initializeExtrapolation();
+
+  /// move to next window and initialize data via extrapolation
+  void moveToNextWindow();
+
+  /// store current value in _extrapolation
+  void storeDataInExtrapolation();
+
 private:
   /**
    * @brief Default constructor, not to be used!
    *
    * Necessary when compiler creates template code for std::map::operator[].
    */
   CouplingData()
-      : requiresInitialization(false)
+      : requiresInitialization(false),
+        _extrapolation(0)
```

Comment:
Wouldn't it be better to use `UNDEFINED_EXTRAPOLATION_ORDER` here as well?

Reply:
Yes. I will implement this :+1: